readNrrdHeader.m

```diff
@@ -0,0 +1,291 @@
+    iLocation = strfind( csU(length('SPACE:')+1:end ), 'RIGHT-ANTERIOR-SUPERIOR'); % only implemented for RAS
+    if ( iLocation>0 )
+      iIsRas = 1;
+    else
+      fprintf('Warning: File is not RAS, make sure subsequent matlab processing is consistent!\n');
+    end
+    
+  elseif ( foundKeyword('SPACE DIRECTIONS:', cs ) )
+    %% Added 'none' for 4D arrays - ofer pasternak 6/3/11
+    iSD = extractNumbersWithout( cs(length('SPACE DIRECTIONS:')+1:end), {'(',')',',','none'} );
+
+    if (length(iSD)~=9)
+      fprintf('Warning: %i space directions found.\n', iSD );
+    end
+    
+    headerInfo.spacedirections = iSD;
+    
+  elseif ( foundKeyword('SIZES:', cs ) )
+    
+    iSizes = sscanf( cs(length('SIZES:')+1:end), '%i' ); % parse sizes
+    headerInfo.sizes = iSizes;
+    
+  elseif ( foundKeyword('THICKNESSES:', cs ) )
+
+    sThicknesses = extractStringList( cs(length('THICKNESSES:')+1:end) );
+    iThicknesses = [];
+    lenThicknesses = length( sThicknesses );
+    for iI=1:lenThicknesses
+      iThicknesses = [iThicknesses, str2num( sThicknesses{iI} ) ];
+    end
+    headerInfo.thicknesses = iThicknesses;
+    
+  elseif ( foundKeyword('KINDS:', cs ) )
+    
+    headerInfo.kinds = extractStringList( cs(length('KINDS:')+1:end) );
+    
+  elseif ( foundKeyword('CENTERINGS:', cs ) )
+    
+    headerInfo.centerings = extractStringList( cs(length('CENTERINGS:')+1:end ) );
+    
+  elseif ( foundKeyword('SPACE UNITS:', cs ) )
+    
+    headerInfo.spaceunits = extractStringList( cs(length('SPACE UNITS:')+1:end ) );
+    
+  elseif ( foundKeyword('SPACE ORIGIN:', cs ) )
+
+    iSO = extractNumbersWithout( cs(length('SPACE ORIGIN:')+1:end), {'(',')',','} );
+    
+    if (length(iSO)~=3)
+      fprintf('Warning: %i space directions found.\n', iSD );
+    end
+    
+    headerInfo.spaceorigin = iSO;
+    
+    
+  elseif ( foundKeyword('MEASUREMENT FRAME:', cs ) )
+    
+    iMF = extractNumbersWithout( cs(length('MEASUREMENT FRAME:')+1:end), {'(',')',','} );
+    headerInfo.measurementframe = iMF;
+    
+  elseif ( foundKeyword('MODALITY', cs ) )
+    
+    headerInfo.modality = removeExcessDelimiters( extractKeyValueString( cs(length('MODALITY')+1:end ) ), ' ');
+    
+  elseif ( foundKeyword('DWMRI_B-VALUE', cs ) )
+    
+    headerInfo.bvalue = str2num( extractKeyValueString( cs(length('DWMRI_B-VALUE')+1:end ) ) );
+    
+  elseif ( foundKeyword('DWMRI_GRADIENT_', cs ) )
+    
+    [iGNr, dwiGradient] = extractGradient( cs(length('DWMRI_GRADIENT_')+1:end ) );
+    headerInfo.gradients(iGNr+1,:) = dwiGradient;
+    
+  else
+    
+    % let's see if we are dealing with a comment
+    
+    csTmp = removeExcessDelimiters( cs, ' ' );
+    if ( csTmp(1)~='#' & ~strcmp(cs(1:4),'NRRD') )
+      fprintf('Warning: Could not parse input line: %s \n', cs );
+    end
+    
+  end
+  
+end
+
+
+fclose( fidr );
+
+
+return
+
+
+function [iGNr, dwiGradient] = extractGradient( st )
+
```

Comment:
Please indent this function 😄

Reply:
The whole readNrrdHeader.m file required indentation. I fixed it.